packages/cli/src/runners/__tests__/task-broker.test.ts

```diff
@@ -102,6 +102,55 @@ describe('TaskBroker', () => {
 
 			expect(runnerIds).toHaveLength(0);
 		});
+
+		it('should remove any pending offers for that runner', () => {
+			const runnerId = 'runner1';
+			const runner = mock<TaskRunner>({ id: runnerId });
+			const messageCallback = jest.fn();
+
+			taskBroker.registerRunner(runner, messageCallback);
+			taskBroker.taskOffered({
+				offerId: 'offer1',
+				runnerId,
+				taskType: 'mock',
+				validFor: 1000,
+				validUntil: createValidUntil(1000),
+			});
+			taskBroker.taskOffered({
+				offerId: 'offer1',
+				runnerId: 'runner2',
+				taskType: 'mock',
+				validFor: 1000,
+				validUntil: createValidUntil(1000),
+			});
+			taskBroker.deregisterRunner(runnerId);
+
+			const offers = taskBroker.getPendingTaskOffers();
+			expect(offers).toHaveLength(1);
+			expect(offers.filter((o) => o.runnerId !== runnerId)).toHaveLength(1);
```

Comment:
This assertion could be more explicit, i.e. assert that the remaining offer is for `runner2`.